raft/raft.go

```diff
@@ -611,6 +611,7 @@ func (r *raft) becomePreCandidate() {
 	// but doesn't change anything else. In particular it does not increase
 	// r.Term or change r.Vote.
 	r.step = stepCandidate
+	r.lead = None
```

Comment:
But some users may use hasLeader() to check whether cluster is in normal state. To let users know StatePreCandidate or inLease is too tricky.

Also in raft paper, it just say:
>
If a follower receives no communication over a period of time called the election timeout, then it assumes there is no viable leader and begins an election to choose a new leader.